Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -249,27 +249,33 @@ protected void SendRPCInternal(Type invokeClass, string rpcName, NetworkWriter w
         [EditorBrowsable(EditorBrowsableState.Never)]
         protected void SendTargetRPCInternal(NetworkConnection conn, Type invokeClass, string rpcName, NetworkWriter writer, int channelId)
         {
-            // this was in Weaver before
             if (!NetworkServer.active)
             {
-                logger.LogError("TargetRPC Function " + rpcName + " called on client.");
+                logger.LogError($"TargetRPC {rpcName} called when server not active");
+                return;
+            }
+
+            if (!isServer)
+            {
+                logger.LogWarning($"TargetRpc {rpcName} called on {name} but that object has not been spawned or has been unspawned");
                 return;
             }
+
             // connection parameter is optional. assign if null.
             if (conn == null)
```

Comment:
@James-Frowen so why do we use '== null' here and 'is null' below?

Reply:
the 2nd if checks if  given parameter or `connectionToClient` are null.  Doing ` if (conn == null)` twice seems a bit odd, thats partly why changed it to `conn = conn ?? connectionToClient;`

Would you rather they both be `== null` or `is null`?

Reply:
it's definitely confusing if == null first and then 'is null' afterwards.
if either of the two versions is better then we should use that version in both cases

Reply:
@vis2k neither one is "better",  they are just different.

if == null checks if the object is truly null or if it has been destroyed.  (slow, it calls the overloaded == operator from unity)
if is null checks if the object is truly null (very fast,  1 single instruction)